code/game/objects/effects/decals/remains.dm

```diff
@@ -18,29 +18,38 @@
 	return !istype(here_turf, /obj/structure/closet/crate/grave/filled) && ..()
 
 /obj/effect/decal/remains/human/smokey
-	desc = "They look like human remains. They have a strange, smokey aura about them..."
+	name = "remains of Charles Morlbaro"
+	desc = "I guess we figured out what happened to the guy who lives here. You'd best tread lightly around this..."
 	///Our proximity monitor, for detecting nearby looters.
 	var/datum/proximity_monitor/proximity_monitor
 	///The reagent we will release when our remains are disturbed.
 	var/datum/reagent/that_shit_that_killed_saddam
+	///A cooldown for how frequently the gas is released when disturbed.
 	COOLDOWN_DECLARE(gas_cooldown)
+	///The length of the aforementioned cooldown.
+	var/gas_cooldown_length = (20 SECONDS)
 
 /obj/effect/decal/remains/human/smokey/Initialize(mapload)
 	. = ..()
 
-	proximity_monitor = new(src, 0)
-	that_shit_that_killed_saddam = get_random_reagent_id()
+	proximity_monitor = new(src, 1)
+	var/list/blocked_reagents = list()
+	for(var/datum/reagent/reagent_path as anything in subtypesof(/datum/reagent))
+		if(ispath(reagent_path, /datum/reagent/medicine) || ispath(reagent_path, /datum/reagent/consumable)) //Boooooriiiiing
+			blocked_reagents += reagent_path
+	that_shit_that_killed_saddam = get_random_reagent_id(blacklist = blocked_reagents)
```

Comment:
You can just use subtypesof on the consumable and medicine typepaths
```suggestion
	var/list/blacklisted_reagents = subtypesof(/datum/reagent/medicine) + subtypesof(/datum/reagent/consumable)
	that_shit_that_killed_saddam = get_random_reagent_id(blacklist = blacklisted_reagents)
```
(you can also put it directly in the proc I don't really care)